File: Leetcode_Medium/_371. Sum of Two Integers/Solution.py
class Solution(object):
    def getSum(self, a, b):
        """
        :type a: int
        :type b: int
        :rtype: int
        """
        while b != 0:
            temp = (a & b) << 1
            a = a ^ b
            b = temp
        return a
        

File: Leetcode_Medium/_371. Sum of Two Integers/Solution.java
class Solution {
    public int getSum(int a, int b) {
        // We're gonna use bit manipulation
        // in this solution
        // We noticed that the bitwise operator
        // OR can yield the same result before 
        // carrying as adding two bit together
        while(b!=0){
            int temp = (a & b) << 1;
            a = a ^ b;
            b = temp;
        }
        return a;
    }
}